exchange/utils.go

```diff
@@ -177,48 +172,79 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 			}
 		}
 
+		ipConf := privacy.IPConf{IPV6: auctionReq.Account.Privacy.IPv6Config, IPV4: auctionReq.Account.Privacy.IPv4Config}
+
+		reqWrapper := cloneBidderReq(bidderRequest.BidRequest)
+
+		// FPD should be applied before policies, otherwise it overrides policies and activities restricted data
+		if auctionReq.FirstPartyData != nil && auctionReq.FirstPartyData[bidderRequest.BidderName] != nil {
+			applyFPD(auctionReq.FirstPartyData[bidderRequest.BidderName], reqWrapper)
+		}
+
 		passIDActivityAllowed := auctionReq.Activities.Allow(privacy.ActivityTransmitUserFPD, scopedName, privacy.NewRequestFromBidRequest(*req))
 		if !passIDActivityAllowed {
-			privacyEnforcement.UFPD = true
+			//UFPD
+			privacy.ScrubDeviceIDs(reqWrapper)
+			privacy.ScrubUserIDs(reqWrapper)
+			privacy.ScrubUserExt(reqWrapper, "data")
 		} else {
 			// run existing policies (GDPR, CCPA, COPPA, LMT)
 			// potentially block passing IDs based on GDPR
 			if gdprEnforced {
 				if gdprErr == nil {
-					privacyEnforcement.GDPRID = !auctionPermissions.PassID
+					if !auctionPermissions.PassID {
+						privacy.ScrubDeviceIDs(reqWrapper)
+						privacy.ScrubUserDemographics(reqWrapper)
+						privacy.ScrubUserExt(reqWrapper, "eids")
```

Comment:
Please consider calling something like `privacy.ScrubGdprID` and move the grouping into scrubber code. Might we be good with calling `ScrubUserFPD` here instead? What would be the difference in information removed?

Reply:
Refactored to the separate functions. It looks cleaner to me when we have separate function per activity in this case, rather than combining then into one, despite they are very similar to each other. I'm open to refactor it to one function.